app/Http/Controllers/BookingController.php

```diff
@@ -188,15 +189,148 @@ public function book($restaurant_id)
+        if ($request->session()->has($user)) {
+            $request->session()->forget($user);
+        }
+        return json_encode(true);
+    }
+
+    private function userSessionKey()
+    {
+        return $user = Auth::user()->id . Auth::user()->username;
+    }
+
+    public function addTables(Request $request)
+    {
+        if (Auth::guest()) {
+            $request->session()->put('redirect_back', URL::previous());
+
+            return redirect('/auth/login');
+        }
+
+        $validator = Validator::make(
+            $request->all(),
+            [
+                'duration' => 'required|digits_between:1,10',
+                'date' => 'required|date_format:d/m/Y H:i',
+            ]
+        );
+
+        if ($validator->fails()) {
+            $output['message'] = "Date and Duration are required.";
+            $output['status'] = 'failure';
+            return json_encode($output);
+        }
+
+        // 14/04/2018 03:00
+
+        // User can only book 30mins from time.
+        // A 2mins delay time is substracted in other to cater for any delay in
+        // tranferring the request to the server.
+        $allowedBookingDateTime = new \DateTime();
+        $allowedBookingDateTime->add(new \DateInterval('PT28M'));
+
+        $bookingDateTime = \DateTimeImmutable::createFromFormat('d/m/Y H:i', $request->date);
+
+        $interval = $allowedBookingDateTime->diff($bookingDateTime);
+
+        // Calculate the seconds difference between the allowedBookingDateTime and bookingDateTime
+        // Reference: http://stackoverflow.com/questions/14277611/convert-dateinterval-object-to-seconds-in-php
+        $seconds = date_create('@0')->add($interval)->getTimestamp();
+
+
+        if ($seconds < 0) {
+            // $output['message'] = "Specify a date at least 30 minutes later.";
+            // $output['status'] = 'failure';
+        } else {
+            $user = Auth::user()->id . Auth::user()->username;
+            $bookedTablesIDs = Session::get($user, function () {
+                return array();
+            });
+
+            $cartData = array();
+            $request->date = str_replace('/', '-', $request->date);
+
+            foreach ($bookedTablesIDs as $bookedTableID) {
+                //Convert to correct format
+                $table = $this->tableRepo->get($bookedTableID);
+
+                $data = $this->prepareCartData($request, $table);
+                array_push($cartData, $data);
+
+                $this->setTableToOnHold($table);
+            }
+
+            $this->addToCart($cartData);
+
+            $this->clearBookingSession();
+
+            $output = [
+                'message'   => 'Tables added to cart.',
+                'status'    => 'success',
+                'tables'    => array_values($bookedTablesIDs),
+            ];
+
+            return json_encode($output);
+        }
     }
 
     public function addTable(Request $request)
     {
+        // return json_encode("hihihhih");
+
```

Comment:
I don't understand this code being here -- this should be handled in the authentication middleware layer at the _routes_.